src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1050,10 +1053,27 @@ private void EmitCatchBlock(BoundCatchBlock catchBlock)
             }
 
             EmitBlock(catchBlock.Body);
-
             _builder.CloseLocalScope();
         }
 
+        private static bool IsFilterConstantFalse(BoundCatchBlock catchBlock)
+        {
+            // Depending on compilation options, BoundLiteral can be wrapped by BoundSequence or BoundSequencePointExpression
+            if(catchBlock.ExceptionFilterOpt != null)
+            {
+                if(catchBlock.ExceptionFilterOpt is BoundSequence sequence)
+                {
+                    return sequence.Value is BoundSequencePointExpression expression
```

Comment:
Hum, this feels strange and makes me think maybe this logic of removing a catch block might be better suited in earlier phases (lowering). @gafter what do you think?